### PR TITLE
Static test

### PR DESCRIPTION
**Pull request by Rayas-NG**

### CODE REVIEW

**Comment by Rayas-NG** [comment]
Fix API issues

File: examples/supplychain-app/quorum/express_nodeJS/controllers/container.js [hunk 1]
```diff
@@ -74,16 +76,18 @@ router.get("/:trackingID?", function(req, res) {
             container.misc[key] = json[key];
           }
 
-
           container.custodian = toPush.custodian;
           container.custodian = container.custodian + "," + toPush.lastScannedAt;
+          container.lastScannedAt = toPush.lastScannedAt,
```
Comment (sownak) [inline examples/supplychain-app/quorum/express_nodeJS/controllers/container.js:81]: ; not , at the end of lines 81 and 83

File: examples/supplychain-app/quorum/express_nodeJS/controllers/container.js [hunk 2]
```diff
@@ -104,12 +108,12 @@ router.post("/", upload.array(), function(req, res) {
     misc: req.body.misc,
     trackingID: req.body.trackingID,
     lastScannedAt: fromNodeSubject,
-    counterparties: req.body.counterparties.map(it =>
-      it.indexOf("O=") != -1 ? it.split("O=")[1].split(",")[0] : it
-    ), //filter out to only send org name
+    counterparties: req.body.counterparties//.map(it =>
+     // it.indexOf("O=") != -1 ? it.split("O=")[1].split(",")[0] : it
```
Comment (sownak) [inline examples/supplychain-app/quorum/express_nodeJS/controllers/container.js:112]: remove commented code if not needed.

File: examples/supplychain-app/quorum/express_nodeJS/controllers/container.js [hunk 3]
```diff
@@ -142,10 +146,6 @@ router.post("/", upload.array(), function(req, res) {
         res.send(error.message);
         console.log(error);
       });
-  } else {
-    res.send(
```
Comment (sownak) [inline examples/supplychain-app/quorum/express_nodeJS/controllers/container.js:146]: this else should not be removed.

File: examples/supplychain-app/quorum/express_nodeJS/controllers/general.js [hunk 4]
```diff
@@ -53,9 +53,10 @@ router.get('/:trackingID/history', function (req, res) {
           .getHistory((i - 1), trackingID)
           .call({ from: fromAddress, gas: 6721975, gasPrice: "0" })
           var history = {};
-          history.custodian = toPush.custodian,
-          history.timestamp = toPush.timestamp,
-          history.lastScannedAt = toPush.lastScannedAt,
+          history.party = toPush.custodian,
```
Comment (sownak) [inline examples/supplychain-app/quorum/express_nodeJS/controllers/general.js:56]: change , to ; for consistency.

File: examples/supplychain-app/quorum/express_nodeJS/controllers/product.js [hunk 5]
```diff
@@ -24,27 +24,30 @@ router.get('/containerless', function (req, res) {
           .getContainerlessAt(i - 1)
           .call({ from: fromAddress, gas: 6721975, gasPrice: "0" })
           var product = {};
+        if(toPush.trackingID != 0){
+          product.trackingID = toPush.trackingID,
```
Comment (sownak) [inline examples/supplychain-app/quorum/express_nodeJS/controllers/product.js:28]: change , to ; for consistency

File: examples/supplychain-app/quorum/express_nodeJS/controllers/product.js [hunk 6]
```diff
@@ -61,7 +64,7 @@ router.get('/:trackingID?', function (req, res) {
       .call({ from: fromAddress, gas: 6721975, gasPrice: "0" })
       .then(response => {
         var newProduct = response;
-        var product = {};
+        var product = {};``
```
Comment (sownak) [inline examples/supplychain-app/quorum/express_nodeJS/controllers/product.js:67]: remove ``

File: examples/supplychain-app/quorum/smartContracts/contracts/ContainerContract.sol [hunk 7]
```diff
@@ -38,7 +38,7 @@ contract ContainerContract is ProductContract {
         string memory _trackingID,
         string memory _lastScannedAt,
         string[] memory _counterparties
-    ) public returns (Container memory) {
+    ) public onlyManufacturer() returns (Container memory) {
```
Comment (sownak) [inline examples/supplychain-app/quorum/smartContracts/contracts/ContainerContract.sol:41]: Can manufacturer only create a container?

File: examples/supplychain-app/quorum/smartContracts/contracts/ContainerContract.sol [hunk 8]
```diff
@@ -84,10 +84,13 @@ contract ContainerContract is ProductContract {
...
         string memory ourAddress = addressToString(msg.sender);
         bool isParticipant = false;
+        string memory lowercaseLongLat = _toLower(_lastScannedAt);
+        bytes memory yourIdentity = abi.encodePacked(ourAddress,",",lowercaseLongLat);
+
 
         for(uint i = 0; i < containerSupplyChain[_containerID].participants.length; i++ ){
             string memory participant = _toLower(containerSupplyChain[_containerID].participants[i]);
-            if(keccak256(abi.encodePacked((ourAddress))) == keccak256(abi.encodePacked((participant))) ) isParticipant = true;
+            if((keccak256(yourIdentity)) == (keccak256(abi.encodePacked((participant)))) ) isParticipant = true;
```
Comment (sownak) [inline examples/supplychain-app/quorum/smartContracts/contracts/ContainerContract.sol:93]: remove one set of ( in encodePacked((participant))

File: examples/supplychain-app/quorum/smartContracts/contracts/ProductContract.sol [hunk 9]
```diff
@@ -127,16 +127,17 @@ function _toLower(string memory str) internal pure returns (string memory) {
...
         string memory ourAddress = addressToString(msg.sender);
         bool isParticipant = false;
         string memory _trackingID;
+        string memory lowercaseLongLat = _toLower(longLat);
+        bytes memory yourIdentity = abi.encodePacked(ourAddress,",",lowercaseLongLat);
 
         for(uint i = 0; i < productSupplyChain[_productID].participants.length; i++ ){
             string memory participant = _toLower(productSupplyChain[_productID].participants[i]);
-            if(keccak256(abi.encodePacked((ourAddress))) == keccak256(abi.encodePacked((participant))) ) {
+            if((keccak256(yourIdentity)) == (keccak256(abi.encodePacked((participant)))) ) {
```
Comment (sownak) [inline examples/supplychain-app/quorum/smartContracts/contracts/ProductContract.sol:140]: remove one set of ( in encodePacked((participant))

File: examples/supplychain-app/quorum/smartContracts/contracts/ContainerContract.sol (hunk 7 above)
Comment (Rayas-NG) [reply examples/supplychain-app/quorum/smartContracts/contracts/ContainerContract.sol:41]: Screen Shot 2020-04-07 at 9 46 05 AM

Comment (Rayas-NG) [reply examples/supplychain-app/quorum/smartContracts/contracts/ContainerContract.sol:41]: Yes. See #291

**Pull request merged**